sssd_test_framework/roles/ipa.py

```diff
@@ -1597,3 +1795,81 @@ def __get_info(self, info: str | NFSExport | IPAAutomountMap | None) -> str | No
             return info.name
 
         return info
+
+class IPAIdView(IPAObject):
+    """
+    IPA id view management.
+    """
+
+    def __init__(self, role:IPA, name:str) -> None:
+        super().__init__(role, name, command_group="idview")
+
+    def add(self,
+            *,
+            description: str | None = None,
+            ) -> IPAIdView:
+        """
+        Create new IPA ID view.
+
+        :return: Self.
+        :rtype: IPAIdView
+        """
+        attrs: CLIBuilderArgs = {
+            "desc": (self.cli.option.VALUE, description),
+        }
+
+        self._add(attrs)
+        return self
+
+
+    def modify(
+        self,
+        *,
+        description: str | None = None,
+        rename: str | None = None,
+    ) -> IPAIdView:
+        """
+        Modify existing IPA ID view.
+
+        Parameters that are not set are ignored.
+
+        :param description: Description, defaults to None
+        :type description: str | None, optional
+        :param rename: Name of IPA ID view, defaults to None
+        :type rename: str | None, optional
+        :return: Self.
+        :rtype: IPAIdView
+        """
+        attrs: CLIBuilderArgs = {
+            "desc": (self.cli.option.VALUE, description),
+            "rename": (self.cli.option.VALUE, rename),
+        }
+
+        self._modify(attrs)
+        return self
+
+    def apply(self, *, hosts: list[str] | None = None, hostgroups: str | None = None) -> IPAIdView:
+        """
+        Applies ID View to specified hosts or current members of specified
```

Comment:
Which id view takes priority, is it the last one applied? I'd put that information in the :description:

```
Applies id view to host or hostgroup. 

:description: ID View is a key. Any overlapping views will replace the value. 
:param hosts: Host or hosts to apply the ID view to, defaults to None.
```
```

Reply:
Updated.